http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -180,15 +216,47 @@ protected String getRedirectUri(String state) {
             if (deployment.getAuthUrl() == null) {
                 return null;
             }
-            URIBuilder redirectUriBuilder = new URIBuilder(deployment.getAuthUrl())
-                    .addParameter(RESPONSE_TYPE, CODE)
-                    .addParameter(CLIENT_ID, deployment.getResourceName())
-                    .addParameter(REDIRECT_URI, rewrittenRedirectUri(url))
-                    .addParameter(STATE, state);
-            redirectUriBuilder.addParameters(forwardedQueryParams);
+
+            String redirectUri = rewrittenRedirectUri(url);
+            URIBuilder redirectUriBuilder = new URIBuilder(deployment.getAuthUrl());
+            redirectUriBuilder.addParameter(RESPONSE_TYPE, CODE)
+                    .addParameter(CLIENT_ID, deployment.getResourceName());
+            if (deployment.getAuthenticationRequestFormat().contains(REQUEST_TYPE_OAUTH2)) {
```

Comment:
We should use an equals check here instead of contains.

Reply:
We could also use a switch statement using the possible enum values instead of the `if...else if...else if`.

Reply:
I also was not quite sure how to go about this because switch statement requires constant value and the way I implemented the Enum, does not have constants. So I just used other constants. 
https://github.com/wildfly-security/wildfly-elytron/blob/6bf8688e3f7bc3df20c757c25178f77d607d836c/http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java#L219-L242
The reason why Default would always be `oauth2` is because in WildFly, I made it so that Authentication-request-format only has those 3 allowed values.